diff --git a/node_modules/isomorphic-git/index.js b/node_modules/isomorphic-git/index.js
index 4c45e45..a098f91 100644
--- a/node_modules/isomorphic-git/index.js
+++ b/node_modules/isomorphic-git/index.js
@@ -3229,6 +3229,7 @@ class GitPackIndex {
     // We need to know the lengths of the slices to compute the CRCs.
     const offsetArray = Object.keys(offsetToObject).map(Number);
     for (const [i, start] of offsetArray.entries()) {
+      if (i % 256 === 0) await new Promise(resolve => setTimeout(resolve, 0));
       const end =
         i + 1 === offsetArray.length ? pack.byteLength - 20 : offsetArray[i + 1];
       const o = offsetToObject[start];
@@ -3265,6 +3266,7 @@ class GitPackIndex {
       }
       count++;
       lastPercent = percent;
+      if (count % 256 === 0) await new Promise(resolve => setTimeout(resolve, 0));
 
       const o = offsetToObject[offset];
       if (o.oid) continue
